File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    answer = True
    if s[-1] == "(" or s[0] == ")":
        return False
    open = 0
    for i in range(len(s)):
        if s[i] == "(":
            open += 1
        else:
            open -= 1
        if open < 0:
            
            return False
    
    if open == 0:
        return True
    else:
        return False

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.js
function solution(s){
    if (s[0] === ")" || s[s.length-1]==="("){
        return false
    }
    let open = 0
    for (let i=0;i<s.length;i++){
        if (s[i] === "("){
            open += 1
        } else {
            open -= 1
        }
        if (open < 0) {
            return false
        }
    }
    if (open !== 0) {
        return false
    }
    else return true;
}